File: Trinity.Runtime/platform_src/android/app/src/main/java/org/elastos/trinity/runtime/passwordmanager/PasswordType.java
package org.elastos.trinity.runtime.passwordmanager;

public enum PasswordType {
    /** Simple password/private key/string info. */
    GENERIC_PASSWORD(0),
    /** Wifi network with SSID and password. */
    WIFI(1),
    /** Bank account, national or international format. */
    BANK_ACCOUNT(2),
    /** Bank card. */
    BANK_CARD(3),
    /** Any kind of account make of an identifier and a password. */
    ACCOUNT(4),
    /** Provider name and key for a 2FA account. */
    TWO_FACTOR_AUTH(5);

    int mValue;

    PasswordType(int value) {
        mValue = value;
    }

    public static PasswordType fromValue(int value) {
        for(PasswordType t : values()) {
            if (t.mValue == value) {
                return t;
            }
        }
        return GENERIC_PASSWORD;
    }
}

File: Trinity.Runtime/platform_src/ios/elastOS/Classes/PasswordManager/PasswordType.swift
/*
* Copyright (c) 2020 Elastos Foundation
*
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the "Software"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
*
* The above copyright notice and this permission notice shall be included in all
* copies or substantial portions of the Software.
*
* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
* SOFTWARE.
*/

import Foundation

public enum PasswordType : Int {
    /** Simple password/private key/string info. */
    case GENERIC_PASSWORD = 0
    /** Wifi network with SSID and password. */
    case WIFI = 1
    /** Bank account, national or international format. */
    case BANK_ACCOUNT = 2
    /** Bank card. */
    case BANK_CARD = 3
    /** Any kind of account make of an identifier and a password. */
    case ACCOUNT = 4
}
